android/src/main/java/com/reactnativecommunity/netinfo/NetworkCallbackConnectivityReceiver.java

```diff
@@ -142,9 +148,13 @@ void updateAndSend() {
 
     private void asyncUpdateAndSend(int delay) {
         new Handler(Looper.getMainLooper()).postDelayed(() -> {
-            mCapabilities = getConnectivityManager().getNetworkCapabilities(mNetwork);
-            updateAndSend();
-
+            try {
+                mCapabilities = getConnectivityManager().getNetworkCapabilities(mNetwork);
+                updateAndSend();
+            } catch (SecurityException e) {
+                // Android 11 may throw a 'package does not belong' security exception here.
+                // We need to catch this to prevent app crash.
```

Comment:
Are these crashes happening because we're missing a permission declaration? https://developer.android.com/reference/android/net/ConnectivityManager#getNetworkCapabilities(android.net.Network)

Reply:
Ah, I see. Maybe add that link in the comment will help others (and our future selves)

Reply:
the formatting from my github web ui code suggestions isn't fantastic but it's a react-native project not a beautiful android java code project, so I'm going to let that go in the interest of (finally!) getting this crash fix out. Thanks for the patience